eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Serialisation/EraIndepGenerators.hs

```diff
@@ -194,29 +140,10 @@ instance Mock c => Arbitrary (BHeader c) where
         bodySize
         bodyHash
 
-instance DSIGNAlgorithm c => Arbitrary (SignedDSIGN c a) where
-  arbitrary =
-    SignedDSIGN . fromJust . rawDeserialiseSigDSIGN
-      <$> (genByteString . fromIntegral $ sizeSigDSIGN (Proxy @c))
-
-instance CC.Crypto c => Arbitrary (BootstrapWitness c) where
-  arbitrary = do
-    key <- arbitrary
-    sig <- genSignature
-    chainCode <- ChainCode <$> arbitrary
-    attributes <- arbitrary
-    pure $ BootstrapWitness key sig chainCode attributes
-
-instance CC.Crypto c => Arbitrary (TP.HashHeader c) where
-  arbitrary = TP.HashHeader <$> genHash
-
-instance (Typeable kr, CC.Crypto c) => Arbitrary (WitVKey kr c) where
-  arbitrary =
-    WitVKey
-      <$> arbitrary
-      <*> arbitrary
+instance Crypto c => Arbitrary (TP.HashHeader c) where
```

Comment:
How come some of these instances stayed here?  What's the criteria?   they are  do only with Shelley maybe?

Reply:
Criteria is that instances for types that are defined in `cardano-ledger-core` should go into `cardano-ledger-core:testlib`. So the `HashHeader`, would need to be migrated into `cardano-protocol-tpraos`, cause that is where it is defined. That should be different PR though. If you feel like it, you could submit such PR, basing on this PR as an exmaple.